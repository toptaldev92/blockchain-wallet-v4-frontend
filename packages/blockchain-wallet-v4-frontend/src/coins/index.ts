import * as features from './features'
import * as selectors from './selectors'
import * as types from './types'
import * as utils from './utils'

export { features, selectors, types, utils }