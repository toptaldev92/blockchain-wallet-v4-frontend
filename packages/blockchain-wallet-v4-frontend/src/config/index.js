import Bitcoin from 'bitcoinjs-lib'
import BitcoinCash from 'bitcoinforksjs-lib'

export default {
  WALLET_DATA_PATH: 'wallet.data',
  WALLET_PAYLOAD_PATH: 'wallet.payload',
  WALLET_SETTINGS_PATH: 'wallet.settings',
  WALLET_OPTIONS_PATH: 'wallet.options',
  WALLET_KVSTORE_PATH: 'wallet.kvstore',
  NETWORK: Bitcoin.networks.bitcoin,
  NETWORK_ETHEREUM: 1,
  NETWORK_BCH: BitcoinCash.networks.bitcoin
}
