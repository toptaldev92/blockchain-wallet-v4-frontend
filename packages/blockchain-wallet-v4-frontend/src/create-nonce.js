/* eslint-disable */
window.__webpack_nonce__ = window.NONCE
/* eslint-enable */
