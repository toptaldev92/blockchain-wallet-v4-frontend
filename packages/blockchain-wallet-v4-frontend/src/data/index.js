import * as actions from './actions'
import * as actionTypes from './actionTypes'
import rootReducer from './rootReducer'
import rootSaga from './rootSaga'
import * as selectors from './selectors'
import * as sagas from './sagas'

export { actions, actionTypes, rootReducer, rootSaga, selectors, sagas }
