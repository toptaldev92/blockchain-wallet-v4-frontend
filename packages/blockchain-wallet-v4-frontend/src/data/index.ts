import * as actions from './actions'
import * as actionTypes from './actionTypes'
import * as model from './model'
import * as sagas from './sagas'
import * as selectors from './selectors'
import rootReducer from './rootReducer'
import rootSaga from './rootSaga'

export { actions, actionTypes, model, rootReducer, rootSaga, selectors, sagas }
