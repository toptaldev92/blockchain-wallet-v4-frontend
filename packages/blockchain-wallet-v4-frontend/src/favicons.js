require('./assets/images/favicon/android-icon-36x36.png')
require('./assets/images/favicon/android-icon-48x48.png')
require('./assets/images/favicon/android-icon-72x72.png')
require('./assets/images/favicon/android-icon-144x144.png')
require('./assets/images/favicon/android-icon-192x192.png')
require('./assets/images/favicon/apple-icon.png')
require('./assets/images/favicon/apple-icon-57x57.png')
require('./assets/images/favicon/apple-icon-60x60.png')
require('./assets/images/favicon/apple-icon-72x72.png')
require('./assets/images/favicon/apple-icon-76x76.png')
require('./assets/images/favicon/apple-icon-114x114.png')
require('./assets/images/favicon/apple-icon-120x120.png')
require('./assets/images/favicon/apple-icon-144x144.png')
require('./assets/images/favicon/apple-icon-152x152.png')
require('./assets/images/favicon/apple-icon-180x180.png')
require('./assets/images/favicon/apple-icon-precomposed.png')
require('./assets/images/favicon/browserconfig.xml')
require('./assets/images/favicon/favicon.ico')
require('./assets/images/favicon/favicon-16x16.png')
require('./assets/images/favicon/favicon-32x32.png')
require('./assets/images/favicon/favicon-96x96.png')
require('./assets/images/favicon/manifest.json')
require('./assets/images/favicon/ms-icon-70x70.png')
require('./assets/images/favicon/ms-icon-144x144.png')
require('./assets/images/favicon/ms-icon-150x150.png')
require('./assets/images/favicon/ms-icon-310x310.png')
require('./assets/images/favicon/site.webmanifest')
