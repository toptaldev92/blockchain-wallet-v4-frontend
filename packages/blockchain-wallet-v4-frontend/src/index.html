<!DOCTYPE html>
<html>
	<head>
		<meta charset="UTF-8" />
		<link rel="apple-touch-icon" sizes="57x57" href="/img/apple-icon-57x57.png">
		<link rel="apple-touch-icon" sizes="60x60" href="/img/apple-icon-60x60.png">
		<link rel="apple-touch-icon" sizes="72x72" href="/img/apple-icon-72x72.png">
		<link rel="apple-touch-icon" sizes="76x76" href="/img/apple-icon-76x76.png">
		<link rel="apple-touch-icon" sizes="114x114" href="/img/apple-icon-114x114.png">
		<link rel="apple-touch-icon" sizes="120x120" href="/img/apple-icon-120x120.png">
		<link rel="apple-touch-icon" sizes="144x144" href="/img/apple-icon-144x144.png">
		<link rel="apple-touch-icon" sizes="152x152" href="/img/apple-icon-152x152.png">
		<link rel="apple-touch-icon" sizes="180x180" href="/img/apple-icon-180x180.png">
		<link rel="icon" type="image/png" sizes="192x192"  href="/img/android-icon-192x192.png">
		<link rel="icon" type="image/png" sizes="32x32" href="/img/favicon-32x32.png">
		<link rel="icon" type="image/png" sizes="96x96" href="/img/favicon-96x96.png">
		<link rel="icon" type="image/png" sizes="16x16" href="/img/favicon-16x16.png">
		<link rel="manifest" href="/img/manifest.json">
		<meta name="msapplication-TileColor" content="#ffffff">
		<meta name="msapplication-TileImage" content="/img/ms-icon-144x144.png">
		<meta name="theme-color" content="#ffffff">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
		<meta name="theme-color" content="#187fc0">
		<meta name="apple-itunes-app" content="app-id=493253309">
		<meta name="description" content="Discover the world's most popular bitcoin wallet. Visit today to create your free simple, secure and safe Blockchain Wallet.">
		<meta name="keywords" content="bitcoin wallet, blockchain wallet, online bitcoin wallet, bitcoin wallet online">
		<title>Blockchain.com Wallet - Exchange Cryptocurrency</title>
		<script nonce="**CSP_NONCE**">
      window.NONCE = '**CSP_NONCE**'
		</script>
	</head>
	<body>
		<div id="app"></div>
	</body>
</html>
