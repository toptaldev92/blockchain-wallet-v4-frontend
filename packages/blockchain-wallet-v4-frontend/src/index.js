__webpack_nonce__ = 'c29tZSBjb29sIHN0cmluZyB3aWxsIHBvcCB1cCAxMjM='
const index =
  process.env.NODE_ENV === 'production'
    ? require('./index.prod')
    : require('./index.dev')

export default index.default
