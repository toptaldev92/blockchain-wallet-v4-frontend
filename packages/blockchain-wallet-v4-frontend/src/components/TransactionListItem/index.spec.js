import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import ListItemContainer from './index'
import configureStore from 'redux-mock-store'

import { Remote } from 'blockchain-wallet-v4/src'
import {
  getOptions,
  getSupportedCoins,
  getErc20CoinList
} from 'blockchain-wallet-v4/src/redux/walletOptions/selectors'

jest.mock('blockchain-wallet-v4/src/redux/walletOptions/selectors')
jest.mock('./template', () => () => <div />)
jest.mock('blockchain-info-components', () => ({
  TooltipRebuild: jest.fn()
}))

getOptions.mockImplementation(() => Remote.of({}))
getSupportedCoins.mockImplementation(() => Remote.of({}))
getErc20CoinList.mockImplementation(() => Remote.of({}))

const store = configureStore([])({})
const tx = { hash: '123abc' }

describe('ListItemContainer', () => {
  it('renders correctly', () => {
    const component = mount(<ListItemContainer store={store} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
  describe('handleEditDescription()', () => {
    it('handles eth tx notes', () => {
      const component = shallow(
        <ListItemContainer transaction={tx} coin='ETH' store={store} />
      )
      const instance = component.dive().instance()
      const spy = jest.spyOn(instance.props.ethActions, 'setTxNotesEth')
      instance.handleEditDescription()
      expect(spy).toHaveBeenCalled()
    })
    it('handles btc tx notes', () => {
      const component = shallow(
        <ListItemContainer transaction={tx} coin='BTC' store={store} />
      )
      const instance = component.dive().instance()
      const spy = jest.spyOn(instance.props.walletActions, 'setTransactionNote')
      instance.handleEditDescription()
      expect(spy).toHaveBeenCalled()
    })
  })
})
