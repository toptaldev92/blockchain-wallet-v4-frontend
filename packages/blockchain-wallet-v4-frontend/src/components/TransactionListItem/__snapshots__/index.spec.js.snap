// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemContainer renders correctly 1`] = `
<Connect(ListItemContainer)
  minConfirmations={3}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ListItemContainer
    ethereumActions={
      Object {
        "createMetadataEthereum": [Function],
        "fetchMetadataEthereum": [Function],
        "fetchMetadataEthereumFailure": [Function],
        "fetchMetadataEthereumLoading": [Function],
        "fetchMetadataEthereumSuccess": [Function],
        "setTxNotesEthereum": [Function],
        "updateMetadataEthereum": [Function],
      }
    }
    minConfirmations={3}
    preferencesActions={
      Object {
        "setBitcoinCashWelcome": [Function],
        "setBitcoinWelcome": [Function],
        "setCulture": [Function],
        "setEtherWelcome": [Function],
        "setLanguage": [Function],
        "setTheme": [Function],
        "toggleCoinDisplayed": [Function],
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    walletActions={
      Object {
        "createLegacyAddress": [Function],
        "deleteHdAddressLabel": [Function],
        "deleteLegacyAddress": [Function],
        "deleteWrapper": [Function],
        "refreshWrapper": [Function],
        "setAccountArchived": [Function],
        "setAccountLabel": [Function],
        "setAddressArchived": [Function],
        "setAutoLogout": [Function],
        "setDefaultAccountIdx": [Function],
        "setHdAddressLabel": [Function],
        "setLegacyAddressLabel": [Function],
        "setMainPassword": [Function],
        "setPayloadChecksum": [Function],
        "setTransactionNote": [Function],
        "setWrapper": [Function],
        "verifyMnemonic": [Function],
      }
    }
  >
    <Component
      handleCoinToggle={[Function]}
      handleEditDescription={[Function]}
      minConfirmations={3}
    >
      <div />
    </Component>
  </ListItemContainer>
</Connect(ListItemContainer)>
`;
